scripts/compile_proto.py

```diff
@@ -378,8 +378,10 @@ def GenerateFunctionDeclarations(self, w):
                 w.Write("void add_%s(%s val);" % (name, cpp_type))
             w.Write("const std::vector<%s>& %s() const;" %
                     (var_cpp_type, name))
+            w.Write("std::vector<%s>* mutable_%s();" % (var_cpp_type, name))
```

Comment:
Note that this breaks the API compatibility with the standard protobuf library (as they have own RepeatedPtrField container instead of a vector, and actually store pointers there instead of values).
I believe it's fine to leave like this though.

Reply:
Is there an easy way to keep compatibility?

Reply:
What probably would work is

1. Create `class RepeatedPtrField` and `class RepeatedField` in `protomessage.h`, which would take std::vector<> as a constructor parameter and wrap it with this https://protobuf.dev/reference/cpp/api-docs/google.protobuf.repeated_field/ interface.
2. Return `RepeatedPtrField` and `RepeatedField` instead of a raw vector.

(not saying it's necessary to do in this PR)

Reply:
(note that the standard protobuf library likely returns the reference to these classes, while we would have to return then by value as they are created on the fly).
Also, in the case of "mutable_" functions, likely it's expected that pointer is returned, which makes it even harder.. Maybe we can approximate it with std::unique_ptr<>..

Reply:
In the end I just added a comment documenting out deviation from the standard protobuf API.